### PR TITLE
Adds nix flake devshell

### PR DESCRIPTION
**Pull request by arpd**
I've only tested this on aarch64 macos, but
should also work on any linux environment.
May have to add `LD_LIBRARY_PATH` in that case.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = (import nixpkgs) { inherit system overlays; };
+      in
+      {
+        # Provide a shell with all dependencies needed to build `koragnar::*`
+        devShell = pkgs.mkShell
+          {
+            nativeBuildInputs = with pkgs; [
+              (rust-bin.selectLatestNightlyWith (toolchain: toolchain.default.override {
```
Comment (vE5li) [inline flake.nix:19]: I was trying to find some refenrence on `selectLatestNightlyWith` to figure out if it respects the `rust-toolchain.toml`, but I believe it does not, correct?

**Review by vE5li — COMMENTED** [review]
Okay first of all, very cool! I've been meaning to add a `flake.nix` for a while but I just didn't know how to do it right. I will probably ask some more questions to fully understand this before merging.

First one right away, does it make a difference if we move these files to the `korangar` directory?
I'm trying to keep the client specific stuff a bit separate from the generic RO stuff, so I would prefer that if possible.

**Comment by arpd** [comment]
I've updated the PR as per your suggestions @vE5li
You can use it either with `direnv`, or `nix develop -c $SHELL` and successfully build and run the project.
It uses the `rust-toolchain.toml` version of the toolchain.

**Comment by vE5li** [comment]
Very nice work! :)

**Pull request merged**